### PR TITLE
[expo-linear-gradient] Avoid unnecessary re-render of NativeLinearGradient on web

### PR DESCRIPTION
**Pull request by danmaas**
# Why

On react-native-web, the NativeLinearGradient component re-renders continuously, eating up CPU cycles.

This seems to be a new behavior with recent versions of `react-native-web`, since I did not notice it before updating from 0.11.7 to 0.13.3.

I believe the problem is that NativeLinearGradient's `onLayout` handler does not check if the layout has actually changed; it just always calls `setLayout()`, which triggers another `onLayout` event, which triggers `setLayout()` again, and so on.

# Test Plan

Use react-native-web 0.13.3, create a LinearGradient, and run a profile using the Chrome React Developer Tools plugin. Without this patch, the NativeLinearGradient component re-renders continuously with the reason "Hooks changed". With the patch, it renders once and then idles properly.

### CODE REVIEW

**Comment by barthap** [comment]
You probably forgot to run `yarn build` from inside `expo-linear-gradient` package to generate js files.

File: packages/expo-linear-gradient/src/NativeLinearGradient.web.tsx [hunk 1]
```diff
@@ -83,7 +83,13 @@ const NativeLinearGradient: FunctionComponent<Props> = ({
         { backgroundImage },
       ]}
       onLayout={event => {
-        setLayout(event.nativeEvent.layout);
+        if (
+          !layout ||
```
Comment (EvanBacon) [inline packages/expo-linear-gradient/src/NativeLinearGradient.web.tsx:87]:
```suggestion
          !layout ??
```

File: packages/expo-linear-gradient/src/NativeLinearGradient.web.tsx [hunk 2]
```diff
@@ -83,7 +83,13 @@ const NativeLinearGradient: FunctionComponent<Props> = ({
...
+          event.nativeEvent.layout.width !== layout.width ||
```
Comment (EvanBacon) [inline packages/expo-linear-gradient/src/NativeLinearGradient.web.tsx:88]: Check x, y as well

**Review by EvanBacon — CHANGES_REQUESTED** [review]
Be sure to ensure the whole object is validated first

**Comment by danmaas** [comment]
Thanks for checking out this PR. I realized I had branched this from an ancient `master` commit, sorry. Fixed now.

I'm not 100% sure I got what you mean by ensuring the whole object is validated first - please check the updated PR and let me know?

**Review by EvanBacon — CHANGES_REQUESTED** [review]
Build files need to be updated (yarn build in packages/expo-linear-gradient) I tried but don't have push permission to your branch

**Comment by danmaas** [comment]
Hmm, it looks like the CI jobs might be referencing an old commit, not the latest commit `cbd9222`, which I force-pushed. I might need to re-generate this PR on a fresh branch, is that OK?

(in the latest commit `cbd9222` I already ran `yarn build` and it includes diffs to `packages/expo-linear-gradient/build/NativeLinearGradient.web.js`).

**Comment by barthap** [comment]
Just do another commit (change something that has no meaning) - it should reset the checks

**Review by EvanBacon — APPROVED** [review]
good work

**Pull request merged**